### PR TITLE
TCharm: Move +vp handling to nodeInit

### PR DESCRIPTION
**Pull request by evan-charmworks**
The old way is a vestige of AMPI multiple worlds, removed in commit
ab97e47ca7fa1819bea179fc4c1cfc2c6676e386.

### CODE REVIEW

**Review by stwhite91 — DISMISSED** [review]

**Review by matthiasdiener — DISMISSED** [review]

**Comment by evan-charmworks** [comment]
I need to test if this is a regression regarding #2536.

**Comment by evan-charmworks** [comment]
PR fixed and result confirmed experimentally.

**Review by stwhite91 — APPROVED** [review]

File: src/libs/ck-libs/tcharm/tcharm.C [hunk 1]
```diff
@@ -71,6 +76,11 @@ void TCharm::nodeInit()
   // Assumes no anytime migration and only static insertion
   _isAnytimeMigration = false;
   _isStaticInsertion = true;
+
+  char **argv = CkGetArgv();
+  nChunks = CkNumPes();
+  CmiGetArgIntDesc(argv,"-vp",&nChunks,"Set the total number of virtual processors");
+  CmiGetArgIntDesc(argv,"+vp",&nChunks,NULL);
```
Comment (matthiasdiener) [inline src/libs/ck-libs/tcharm/tcharm.C:83]:
```suggestion
  CmiGetArgIntDesc(argv, "-vp", &nChunks, "Set the total number of virtual processors");
  CmiGetArgIntDesc(argv, "+vp", &nChunks, NULL);
```

File: src/libs/ck-libs/tcharm/tcharm.C [hunk 2]
```diff
@@ -11,7 +11,7 @@ Orion Sky Lawlor, olawlor@acm.org, 11/19/2001
 
 CtvDeclare(TCharm *,_curTCharm);
 
-static int lastNumChunks=0;
+static int nChunks;
```
Comment (matthiasdiener) [inline src/libs/ck-libs/tcharm/tcharm.C:14]: Any particular reason why this variable and the static variable below are no initialized?

Comment (evan-charmworks) [reply src/libs/ck-libs/tcharm/tcharm.C:14]: They get implicitly set to zero by the BSS segment.

Comment (matthiasdiener) [reply src/libs/ck-libs/tcharm/tcharm.C:14]: Sure, but why not write it down explicitly?

Comment (evan-charmworks) [reply src/libs/ck-libs/tcharm/tcharm.C:14]: Doesn't that increase the size of the .data segment?

Comment (matthiasdiener) [reply src/libs/ck-libs/tcharm/tcharm.C:14]: I think it does, but no idea if that's important or not. This PR is fine for me either way.

**Review by matthiasdiener — APPROVED** [review]

**Review by stwhite91 — APPROVED** [review]

**Pull request merged**